scanners/flutter/flutter.go

```diff
@@ -0,0 +1,261 @@
+	fileList, err := utility.ListPathInDirSortedByComponents(projectLocation, true)
+	if err != nil {
+		return nil, err
+	}
+
+	for i, file := range fileList {
+		fileList[i] = filepath.Join(projectLocation, file)
+	}
+
+	filters := []utility.FilterFunc{
+		ios.AllowXCWorkspaceExtFilter,
+		ios.AllowIsDirectoryFilter,
+		ios.ForbidEmbeddedWorkspaceRegexpFilter,
+		ios.ForbidGitDirComponentFilter,
+		ios.ForbidPodsDirComponentFilter,
+		ios.ForbidCarthageDirComponentFilter,
+		ios.ForbidFramworkComponentWithExtensionFilter,
+		ios.ForbidCordovaLibDirComponentFilter,
+		ios.ForbidNodeModulesComponentFilter,
+	}
+
+	return utility.FilterPaths(fileList, filters...)
+}
+
+// DetectPlatform ...
+func (scanner *Scanner) DetectPlatform(searchDir string) (bool, error) {
+	projectLocations, err := findProjectLocations(searchDir)
+	if err != nil {
+		return false, err
+	}
+
+	for _, projectLocation := range projectLocations {
+		workspaceLocations, err := findWorkspaceLocations(projectLocation)
+		if err != nil {
+			return false, err
+		}
+
+		for _, workspaceLocation := range workspaceLocations {
+			ws, err := xcworkspace.Open(workspaceLocation)
+			if err != nil {
+				return false, nil
+			}
+			schemeMap, err := ws.Schemes()
+			if err != nil {
+				return false, nil
+			}
+
+			proj := project{
+				path:              projectLocation,
+				xcodeProjectPaths: map[string][]string{},
+			}
+
+			for _, schemes := range schemeMap {
+				for _, scheme := range schemes {
+					proj.xcodeProjectPaths[workspaceLocation] = append(proj.xcodeProjectPaths[workspaceLocation], scheme.Name)
+				}
+			}
+
+			scanner.projects = append(scanner.projects, proj)
+		}
+	}
+
+	if len(scanner.projects) == 0 {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+// ExcludedScannerNames ...
+func (Scanner) ExcludedScannerNames() []string {
+	return []string{
+		string(ios.XcodeProjectTypeIOS),
+		string(ios.XcodeProjectTypeMacOS),
+		cordova.ScannerName,
+		android.ScannerName,
+	}
+}
+
+// Options ...
+func (scanner *Scanner) Options() (models.OptionModel, models.Warnings, error) {
+	flutterProjectLocationOption := models.NewOption(projectLocationInputTitle, projectLocationInputEnvKey)
+
+	for _, project := range scanner.projects {
+		projectPathOption := models.NewOption(ios.ProjectPathInputTitle, ios.ProjectPathInputEnvKey)
+		flutterProjectLocationOption.AddOption(project.path, projectPathOption)
+
+		for xcodeWorkspacePath, schemes := range project.xcodeProjectPaths {
+			schemeOption := models.NewOption(ios.SchemeInputTitle, ios.SchemeInputEnvKey)
+			projectPathOption.AddOption(xcodeWorkspacePath, schemeOption)
+
+			for _, scheme := range schemes {
+				exportMethodOption := models.NewOption(ios.IosExportMethodInputTitle, ios.ExportMethodInputEnvKey)
+				schemeOption.AddOption(scheme, exportMethodOption)
+
+				for _, exportMethod := range ios.IosExportMethods {
+					configOption := models.NewConfigOption(defaultConfigName)
```

Comment:
is it intended: using the same config name for default and scanned workflows?

Reply:
i see so the default and scanned configs are the same, so they share a name, this case my i would not add the default suffix to the workflow name